### PR TITLE
[helm] add third scalable target `backend`

### PR DESCRIPTION
**Pull request by trevorwhitney**
**What this PR does / why we need it**:

Adds the new 3rd scalable deployment mode target (backend) to the helm chart. This new target was added in #7650

**Special notes for your reviewer**:

This PR also adds a new loki dev env in `tools/dev/k3d`. It's currently hardcoding to a Loki image that has this target in it so I can test this new mode. We can change that after the next release or weekly to point to something a bit more canonical.

**Checklist**
- [ ] Reviewed the `CONTRIBUTING.md` guide
- [ ] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

File: production/helm/loki/templates/read/deployment-read.yaml [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+      securityContext:
+        {{- toYaml .Values.loki.podSecurityContext | nindent 8 }}
+      terminationGracePeriodSeconds: {{ .Values.read.terminationGracePeriodSeconds }}
+      containers:
+        - name: read
+          image: {{ include "loki.image" . }}
+          imagePullPolicy: {{ .Values.loki.image.pullPolicy }}
+          args:
+            - -config.file=/etc/loki/config/config.yaml
+            - -target={{ .Values.read.targetModule }}
```
Comment (jeschkies) [inline production/helm/loki/templates/read/deployment-read.yaml:56]: Where is this defined?

File: production/helm/loki/templates/backend/service-backend-headless.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{{- $isSimpleScalable := eq (include "loki.deployment.isScalable" .) "true" -}}
+{{- if and $isSimpleScalable (not .Values.read.legacyReadTarget ) }}
+---
+apiVersion: v1
+kind: Service
```
Comment (jeschkies) [inline production/helm/loki/templates/backend/service-backend-headless.yaml:5]: Why is this needed? I see a service below. Is there some duplication?

**Review by jeschkies — COMMENTED** [review]
Nice to see this happening. How would one migrate from Helm Chart 3.x?

File: production/helm/loki/templates/backend/service-backend-headless.yaml (hunk 2 above)
Comment (trevorwhitney) [reply production/helm/loki/templates/backend/service-backend-headless.yaml:5]: I don't know k8s well enough to answer this, but I'm following the pattern of our other statefulsets that have both a service and a headless service. The headless service is used as the statefulset's `spec.serviceName`, whereas the other service is used for load-balancing.

File: production/helm/loki/templates/read/deployment-read.yaml (hunk 1 above)
Comment (trevorwhitney) [reply production/helm/loki/templates/read/deployment-read.yaml:56]: https://github.com/trevorwhitney/loki/blob/0621044bcb4df2388721e604ff46b61e22506529/production/helm/loki/values.yaml#L825

**Review by jeschkies — CHANGES_REQUESTED** [review]
I think this is good to go. However, what is the migration process from 3.x? What documentation should we add?

**Comment by trevorwhitney** [comment]
@jeschkies migration guide added
